Sources/NIOHTTP1/HTTPDecoder.swift

```diff
@@ -234,6 +237,23 @@ public class HTTPDecoder<HTTPMessageT>: ByteToMessageDecoder, AnyHTTPDecoder {
         return response
     }
 
+    private func bytesToForwardOnRemoval(ctx: ChannelHandlerContext) -> ByteBuffer? {
+        guard self.parser.upgrade == 1 && ctx.channel.isActive else {
+            return nil
+        }
+        if let buffer = self.cumulationBuffer?.slice(), buffer.readableBytes > 0 {
```

Comment:
Out of interest, why are we slicing the cumulation buffer here?

Reply:
@Lukasa I just thought it may be a bit nicer to consume. That said we not need to slice it. Happy to remove if you think we should

Reply:
Unexplained slicing without a code comment is the kind of thing that we'll be real nervous to change in 6 months time. :D I think either remove it or put a comment explaining why it's there.

Reply:
added a comment ;)